examples/filters_example.cpp

```diff
@@ -0,0 +1,327 @@
+            ImGui::TreePush();
+            float w = ImGui::GetContentRegionAvailWidth();
+            float p = ImGui::GetStyle().FramePadding.x;
+            if (ImGui::Button("Load##PointCloud", ImVec2((w - p) / 2.f, 0)) && !is_filter_running())
+            {
+                reset();
+                std::string const filename = igl::file_dialog_open();
+                std::filesystem::path ply_point_cloud{filename};
+                auto [p, _]  = pcp::io::read_ply<point_type, normal_type>(ply_point_cloud);
+                points       = std::move(p);
+                auto const V = from_point_cloud(points);
+                viewer.data().clear();
+                viewer.data().add_points(V, Eigen::RowVector3d(1.0, 1.0, 0.0));
+                viewer.data().point_size = 1.f;
+                viewer.core().align_camera_center(V);
+            }
+
+            ImGui::SameLine();
+            if (ImGui::Button("Save##PointCloud", ImVec2((w - p) / 2.f, 0.f)))
+            {
+                std::filesystem::path ply_mesh = igl::file_dialog_save();
+                pcp::io::write_ply(
+                    ply_mesh,
+                    points,
+                    std::vector<normal_type>{},
+                    pcp::io::ply_format_t::binary_little_endian);
+            }
+
+            static ImU64 const step = 1;
+
+            if (ImGui::CollapsingHeader("Display##PointCloud", ImGuiTreeNodeFlags_DefaultOpen))
+            {
+                std::string const points_str =
+                    std::string("Points: ") + std::to_string(points.size());
+                ImGui::Text(points_str.c_str());
+            }
+
+            if (ImGui::CollapsingHeader("Filters", ImGuiTreeNodeFlags_DefaultOpen))
+            {
+                ImGui::RadioButton("statistical", &filters_variant, 0);
+
+                ImGui::InputFloat(
+                    "Standard dev multiplier threshold",
+                    &std_dev_multiplier_threshold,
+                    0.1f,
+                    1.0f,
+                    "%0.2f");
+
+                ImGui::InputScalar("mean k", ImGuiDataType_U64, &mean_k, &step);
+
+                ImGui::RadioButton("pass through", &filters_variant, 1);
+
+                ImGui::InputScalar("index", ImGuiDataType_U64, &index_to_filter, &step);
+                ImGui::InputFloat("min", &filter_limit_min, 0.1f, 1.0f, "%0.2f");
+                ImGui::InputFloat("max", &filter_limit_max, 0.1f, 1.0f, "%0.2f");
+
+                ImGui::RadioButton("radius outlier", &filters_variant, 2);
+
+                ImGui::InputFloat("radius", &radius, 0.1f, 1.0f, "%0.2f");
+                ImGui::InputScalar(
+                    "min neighbors in radius",
+                    ImGuiDataType_U64,
+                    &min_neighbors_in_radius,
+                    &step);
+
+                ImGui::RadioButton("density", &filters_variant, 3);
+
+                ImGui::InputFloat("density threshold", &density_threshold, 0.1f, 1.0f, "%0.2f");
```

Comment:
Add some text to specify what density metric we are using, exactly. Is this number of neighbors in a ball, is this a measure of the mean distance to neighbours vs global mean distance, etc.?